panther_analysis_tool/lookup_table.py

```diff
@@ -0,0 +1,211 @@
+"""
+Panther Analysis Tool is a command line interface for writing,
+testing, and packaging policies/rules.
+Copyright (C) 2020 Panther Labs Inc
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import os
+import hashlib
+import json
+import logging
+from urllib import parse
+
+import boto3
+import requests
+
+from panther_analysis_tool.util import get_client
+
+LOOKUP_LAMBDA_NAME = "panther-lookup-tables-api"
+UPLOAD_PREFIX = "luts"
+
+class LookupTable():
+    """A Panther Lookup Table"""
+
+    def __init__(self, spec: dict, spec_dir: str, aws_profile: str):
+        """Instantiate a Lookup Table object
+
+        Args:
+            spec: a dict representing the configuration of a Lookup Table
+        """
+        self.name = spec["name"]
+        self.spec = spec
+        self.spec_dir = spec_dir
+        self.aws_profile = aws_profile
+        self.client = get_client(self.aws_profile, "lambda")
+        self.lookup_id = None
+        self.input_bucket = ""
+        logging.info("Creating/updating the lookup table %s", self.name)
+
+
+    def create_or_update(self) -> int:
```

Comment:
`[minor]` what do you think about naming this upsert instead?

Reply:
@darwayne There's another file which has an `update_or_create` method so this felt in line with that style. Doesn't matter too much to me either way. Doesn't really feel like an insert/update/upsert since we're not talking to database directly.

But your team owns this repo so tell me what you want. :)

Reply:
@matthewcummings got it. Anytime I see a function that is only responsible for creating or updating I recommend naming it upsert since it is concise an encapsulates both create and update. create is synonymous to insert. So upsert doesn't necessarily imply what your underlying storage mechanism is.